sanitycheck: a fix and an enhancement to hw.yml validation

pabigot opened the pull request:
A legitimate optional key was not included, breaking some hardware yaml files.

The change to validation removed the ability to annotate devices; add a key to restore this functionality.

pabigot committed:
sanitycheck: add missing probe_id key to schema

probe_id is used as an optional key that overrides id as the source
of board_id when configured to use a runner that is not the default.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

pabigot committed:
sanitycheck: add notes key to schema

Prior to addition of schema validation running sanitycheck
--generate-hw-map would preserve the value of any field in an existing
map that it didn't update.  These fields now cause a parsing error.
Add notes as an optional key where information relevant to the board
can be added to provide useful context (such as which serial port
corresponds to the console, or why a non-standard runner is required).

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

nashif commented:
> A legitimate optional key was not included, breaking some hardware yaml files.

where was this added and what key are you referring to?

pabigot commented:
> > A legitimate optional key was not included, breaking some hardware yaml files.
> 
> where was this added and what key are you referring to?

da7384846e74d86568db14a9950d779e85f87119 and `probe_id`.

nashif commented:
> [da73848](https://github.com/zephyrproject-rtos/zephyr/commit/da7384846e74d86568db14a9950d779e85f87119) and `probe_id`.

ok, and what is "notes"?

Btw, can document probe_id while at it.

pabigot commented:
The role of `notes` is described in the commit message.

I don't see documentation on any of the fields in the hardware map except `fixtures`.  Do you want a section each for `probe-id` and `notes`?  Or something more encompassing?

nashif commented:
> I don't see documentation on any of the fields in the hardware map except `fixtures`.

well, I thought those are self-explanatory :), probe_id however has some special handling as it overrides. Anyways, I will update the docs with more details, you can add notes and probe_id now, basically the content of the commit message here and what you had in da7384846e74d86568db14a9950d779e85f87119.

Thanks.

pabigot committed:
doc: guides: sanity_chk: add documentation on hardware map keyword

The probe_id keyword was added to support overriding the inferred id
when determining the board-id to use when flashing.  Document the
keyword.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>
